File: Methodo/Perl/puissance.pl
#!usr/bin/perl -w 
sub puissance {
	$result = $x;
				if($n == 0){
							$result = 1;
							print("$x puiss $n  = $result\n");
							if($n == 1){
										$result = $x;
										print("$x puiss $n = $result\n");
							}
				}
				else{
					for($i = $n; $i > 1; $i--){ 
												$result = $result * $x;
					}
				print("$x puiss $n = $result\n");
				}
}
print("entrez x :\n");
$x = <STDIN>; chomp $x;
print("puissance ? \n");
$n = <STDIN>; chomp $n;
puissance();

File: Methodo/C/Calcul/puissance.c
#include <stdio.h>

int puissance(int x, int n);


int main(){

	int x, n;
	printf("nombre ? ");
	scanf("%d", &x);
	printf("puissance ? ");
	scanf("%d", &n);

	printf("%d puiss(%d)=%d\n", x, n, puissance(x,n));

	return 0;
}

int puissance(int x, int n){
	if(n == 0){
		return 1;
		if(n == 1){
			return x;
		}
	}
	else{
		return x * puissance(x, n - 1);
	}
}